patterns/ecs-v1/grok-patterns

```diff
@@ -80,16 +80,16 @@ DATESTAMP_EVENTLOG %{YEAR}%{MONTHNUM2}%{MONTHDAY}%{HOUR}%{MINUTE}%{SECOND}
 # Syslog Dates: Month Day HH:MM:SS
 SYSLOGTIMESTAMP %{MONTH} +%{MONTHDAY} %{TIME}
 PROG [\x21-\x5a\x5c\x5e-\x7e]+
-SYSLOGPROG %{PROG:program}(?:\[%{POSINT:pid}\])?
+SYSLOGPROG %{PROG:[process][name]}(?:\[%{POSINT:[process][pid]:int}\])?
 SYSLOGHOST %{IPORHOST}
-SYSLOGFACILITY <%{NONNEGINT:facility}.%{NONNEGINT:priority}>
+SYSLOGFACILITY <%{NONNEGINT:[log][syslog][facility][code]:int}.%{NONNEGINT:[log][syslog][priority]:int}>
 HTTPDATE %{MONTHDAY}/%{MONTH}/%{YEAR}:%{TIME} %{INT}
 
 # Shortcuts
 QS %{QUOTEDSTRING}
 
 # Log formats
-SYSLOGBASE %{SYSLOGTIMESTAMP:timestamp} (?:%{SYSLOGFACILITY} )?%{SYSLOGHOST:logsource} %{SYSLOGPROG}:
+SYSLOGBASE %{SYSLOGTIMESTAMP:timestamp} (?:%{SYSLOGFACILITY} )?%{SYSLOGHOST:[host][hostname]} %{SYSLOGPROG}:
```

Comment:
Quick note on `host.name` vs `host.hostname`. This may or may not change how you approach the parsing here.

`host.hostname` is meant to be the hostname as is reported by the system or syslog.

`host.name` is meant to default to the hostname, but be configurable. This helps in situations like managing a fleet of Macs mostly named "MacBook-Pro" :-)

The Elastic Security app uses `host.name` is the primary identifier for hosts. If `host.name` is not populated, events won't show up in SIEM, or there will be problems.

So I think mapping this to `host.hostname` is correct in the context of these Syslog groks.

But perhaps we should mention the subtlety above in some Logstash documentation?

Reply:
I've setup a draft for the syslog input, that might be a good place to put this so I put :scissors: your notes there as a reminder: https://github.com/logstash-plugins/logstash-input-syslog/pull/63#issuecomment-663156390